test/oauth2c_token_cache_gen_server_SUITE.erl

```diff
@@ -0,0 +1,45 @@
+-module(oauth2c_token_cache_gen_server_SUITE).
+-compile([export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+
+all() -> [
+  get_token,
+  insert_token,
+  delete_token
+].
+
+init_per_suite(Config) -> Config.
+end_per_suite(_Config) -> ok.
+
+init_per_testcase(_TestCase, Config) ->
+  Now = os:system_time(millisecond),
+  TTL = 3.6e6,
+  InitialState = #{cache =>
+    #{key1 => {token1, Now}, key2 => {token2, Now - TTL}}, cache_ttl => TTL},
```

Comment:
Since these are actually only used in the `get_token` test case, they could be in a clause specific to this test.

Reply:
I understand your point. However, at the moment it is only possible to inject an initial state when the server is started (i.e. in the start_link function).  For instance, it will be impossible to test the expiration of tokens, if I do not inject an initial state where I can specify exactly the creation time of the tokens because the creation time is otherwise automatically set by the server when you are using the insert method.  Do you have any suggestions of how to solve this?

Reply:
Note that `init_per_test_case` takes the test case as a first parameter. A common pattern is to write the `init_per_test_case` as:
```
init_per_test_case(TestCase, Config) ->
  <General initialization>
  tc_init(TestCase, Config).

tc_init(test_case_1, Config) ->
  <specific init for test_case_1>;
tc_init(test_case_2, Config) ->
  <specific init for test_case_2>;
...
tc_init(_TestCase, Config) ->
  Config.
```
And the same for `end_per_test_case`.

A variation of this that keeps the init and tear-down close to the test case is one that is used in several applications in Kivra (eg., `kivra_core`). In this you have:
```
init_per_testcase(TestCase, Config) ->
    ?MODULE:TestCase({init, Config}).

end_per_testcase(TestCase, Config) ->
    ?MODULE:TestCase({'end', Config}).
```
and you write each test case as:
```
test_case_1({init, Config}) ->
  <specific init for test_case_1>;
test_case_1({'end', Config}) ->
  <specific tear down for test_case_1>;
test_case_1(Config) ->
  <the actual test>.
```

Reply:
I have changed my common tests in accordance with your recommendations!